algebird-test/src/test/scala/com/twitter/algebird/macros/CuberMacroTest.scala

```diff
@@ -0,0 +1,67 @@
+package com.twitter.algebird.macros
+
+import com.twitter.algebird._
+import com.twitter.algebird.macros._
+import com.twitter.algebird.macros.ArbitraryCaseClassMacro.arbitrary
+
+import org.scalacheck.{ Properties, Arbitrary }
+import org.scalacheck.Prop.forAll
+
+object CuberRollerProperties extends Properties("Cuber and roller macro") {
+  import BaseProperties._
+
+  implicit val arbitraryFoo: Arbitrary[Foo] = arbitrary[Foo]
+  implicit val arbitraryBar: Arbitrary[Bar] = arbitrary[Bar]
+  implicit val arbitraryBaz: Arbitrary[Baz] = arbitrary[Baz]
+
+  case class Foo(a: Int, b: Option[String], c: Long)
```

Comment:
can we add a test case here of a nested tuple or case class to show it not recursing